Cargo.toml

```diff
@@ -52,6 +52,10 @@ rt = ["stm32f3/rt"]
 stm32f301 = ["stm32f3/stm32f301", "device-selected"]
 stm32f318 = ["stm32f3/stm32f3x8", "device-selected"]
 stm32f302 = ["stm32f3/stm32f302", "device-selected"]
```

Comment:
With the changes in clock calculation, this would implicitly "choose" stm32f302xb / stm32f302xc for the clock calculation. Is this desired behavior? Do we want to force the user to specify the device if needed (i assume this will also happen for other devices in the future) or do we implicitly choose one? This could lead to unexpected results for the user as some registers could be set to wrong states.

Reply:
I'd vouch for forcing the user to specify the device (i.e. removing "device-selected" from stm32f303 and adding it to all the sub-variants). 
I expect 
* Less people getting confused (see e.g. https://github.com/stm32-rs/stm32f3xx-hal/issues/62)
* Some backwards comparability getting broken (but that's fine)
* More awareness when implementing stuff
* More stuff getting implemented (as it is easier to adopt something for a sub-variant if there is already the infrastructure for selecting only this variant)

It'd like to see
* a throughout instructions in the README on how to select the proper decive
* a short note to read the mentioned README instructions when [no sub-variant is used](https://github.com/stm32-rs/stm32f3xx-hal/blob/14ed7e55ed8f83f09a54d08fc81d7575d6404403/src/lib.rs#L5) 
(side note: I proposed something along those lines in https://github.com/stm32-rs/stm32f3xx-hal/pull/47 - see [the errormsg there](https://github.com/strom-und-spiele/stm32f3xx-hal/blob/fb5d6a7a1aa5d6b31eedbc7ac434f5e4bf1010a5/src/lib.rs#L5) )

Reply:
Good point. Confusing users when falling back implicitly to a specific device is definitely not the way to go. We should handle every sub-variant as and independent device. This would make some parts of this HAL more verbose, but then it's easier to reason about little differences between those devices.

I'd do this with a cleanup of the README in a follow up PR, as it would not hurt, that the README can get a bit more information.

Reply:
Actually we don't need to be more verbose. See my proposal in https://github.com/stm32-rs/stm32f3xx-hal/pull/75